### PR TITLE
Introduce next.js for migration to React.

### PR DESCRIPTION
**Pull request by hblumberg**
@snyderc, @mwellman17, and I are hoping to migrate the frontend to React! Chris found that [Next.js](https://nextjs.org) will allow us to serve React pages alongside our existing EJS pages and Express endpoints. Next also allows for server-side rendering for better SEO and has internationalization support that we might be able to use.

This first change adds the necessary React and Next dependencies and adds a temporary React page as a proof of concept. Going forward, we'll actually start migrating to React!

### Test Plan:
* Ensure the whole website functions as usual (with no console errors).
* Double check that the search functionality works as expected.
* Go to localhost:3002/dev and see "Hello, world!".

Co-authored-by: Chris Snyder <christopher.snyder.m@gmail.com>

### CODE REVIEW

File: pages/dev.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+// expected. Once we've migrated over, remove this!
+const Dev = () => {
+    return(
+      <div>
+          Hello, world!
+      </div>
+    );
+  }
+  
+export default Dev;
```
Comment (hblumberg) [inline pages/dev.js:11]: I'm seeing (inconsequential) console errors on this page that I have to investigate, but putting the PR up for feedback in the meantime!

**Comment by johndeyrup** [comment]
Seems reasonable

**Comment by mzagaja** [comment]
I haven't used Next.js before but have worked on a bunch of React projects. The Next.js documentation looks like it is pretty good, it appears this library has been widely adopted, and I appreciate that it has been designed for gradual adoption. Finally, given I think a lot of folks have experience with React, I am supportive of this move.

File: pages/dev.js (hunk 1 above)
Comment (hblumberg) [reply pages/dev.js:11]: Fixed!

**Comment by hblumberg** [comment]
@johndeyrup and @mzagaja -- thanks for the quick reviews! I fixed the console errors on the new React page if you want to take another look (: If not, I'll aim to deploy later today!

**Comment by johndeyrup** [comment]
Feel free to :shipit:

**Pull request merged**